pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/GrpcSendingMailbox.java

```diff
@@ -53,7 +54,7 @@ public GrpcSendingMailbox(String mailboxId, GrpcMailboxService mailboxService) {
   public void init()
       throws UnsupportedOperationException {
     ManagedChannel channel = _mailboxService.getChannel(_mailboxId);
-    PinotMailboxGrpc.PinotMailboxStub stub = PinotMailboxGrpc.newStub(channel);
+    PinotMailboxGrpc.PinotMailboxStub stub = PinotMailboxGrpc.newStub(channel).withDeadlineAfter(2, TimeUnit.MINUTES);
```

Comment:
Can we pass in the query deadline instead of an arbitrary deadline? Otherwise, for query that takes 2+ minutes, this deadline doesn't work.

Reply:
+1 let's use the query deadline here as it is a a GRPC stub deadline not a mailbox concept deadline